engine/language-server/src/main/scala/org/enso/languageserver/libraries/ComponentGroupsValidator.scala

```diff
@@ -45,7 +45,8 @@ final class ComponentGroupsValidator {
   ): Map[LibraryName, Either[ValidationError, ComponentGroups]] = {
     val init: Map[LibraryName, Either[ValidationError, ComponentGroups]] =
       componentGroups.map { case (k, v) => k -> Right(v) }
-    val groupsMap: mutable.Map[GroupReference, ComponentGroup] = mutable.Map()
+    val groupsMap: mutable.Map[GroupReference, ComponentGroup] =
+      mutable.HashMap()
```

Comment:
Is `HashMap` appropriate here? I assume it cannot guarantee order, but I am not sure whether keeping order is needed here.

Reply:
Hash map here doesn't break the order

Reply:
According to [stackoverflow](https://stackoverflow.com/questions/10113448/scala-hashmap-iterate-in-insertion-order) _In general, a Scala HashMap does not guarantee the original order._.